src/main/java/mesosphere/marathon/metrics/Metrics.scala

```diff
@@ -65,6 +66,16 @@ class Metrics @Inject() (val registry: MetricRegistry) {
 
 object Metrics {
   class Timer(timer: com.codahale.metrics.Timer) {
+    def futureSuccess[T](future: => Future[T]): Future[T] = {
+      val startTime = System.nanoTime()
+      import scala.concurrent.ExecutionContext.Implicits.global
+      val f = future
+      f.onSuccess {
```

Comment:
Just curious - why only on success?
Because exceptions will distort the times?
My take: it would be interesting, how often the future is called and what the mean time to execute is, even in failure case.